packages/actor-query-operation-path-zero-or-one/lib/ActorQueryOperationPathZeroOrOne.ts

```diff
@@ -36,6 +37,21 @@ export class ActorQueryOperationPathZeroOrOne extends ActorAbstractPath {
       throw new Error('ZeroOrOne path expressions with 2 variables not supported yet');
     }
 
+    // Such connectivity matching does not introduce duplicates (it does not incorporate any count of the number
+    // of ways the connection can be made) even if the repeated path itself would otherwise result in duplicates.
+    // https://www.w3.org/TR/sparql11-query/#propertypaths
+    if (!context || !context.get('isPathArbitraryLengthDistinct')) {
```

Comment:
Could you abstract this block into the abstract path actor? Because it seems so occur more than once. (Copy-pasting is bad for maintainability)